MathLib/LinAlg/Lis/LisMatrix.cpp

```diff
@@ -87,6 +87,7 @@ void LisMatrix::setZero()
 
 int LisMatrix::setValue(IndexType rowId, IndexType colId, double v)
 {
+    if (v == 0.0) return 0;
```

Comment:
It does not influence the generation of the sparsity pattern, only its application:
In general the sparsity pattern now will be slightly too large, so more space than needed will be reserved. However, I think we won't run out of space because of that.
`if (v==0.0)` will also not influence subsequent iterations, because currently the method `setZero()` of LisMatrix rebuilds the whole matrix and the sparsity pattern has to be applied in every step.